internal/handshake/crypto_setup.go

```diff
@@ -541,13 +542,9 @@ func (h *cryptoSetup) GetOpener(level protocol.EncryptionLevel) (Opener, error)
 	}
 }
 
-func (h *cryptoSetup) ConnectionState() ConnectionState {
-	connState := h.conn.ConnectionState()
-	return ConnectionState{
-		HandshakeComplete: connState.HandshakeComplete,
-		ServerName:        connState.ServerName,
-		PeerCertificates:  connState.PeerCertificates,
-	}
+func (h *cryptoSetup) ConnectionState() tls.ConnectionState {
+	cs := h.conn.ConnectionState()
+	return *(*tls.ConnectionState)(unsafe.Pointer(&cs))
```

Comment:
Add a comment here explaining the rationale, and why this is the best option.

Many people audit uses of "unsafe", and it's very helpful to have local docs on that.